kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/PurchaseOrderAmendmentDocument.java

```diff
@@ -0,0 +1,49 @@
+package edu.arizona.kfs.module.purap.document;
+
+import org.kuali.kfs.sys.businessobject.Building;
+import edu.arizona.kfs.sys.businessobject.BuildingExtension;
+
+public class PurchaseOrderAmendmentDocument extends org.kuali.kfs.module.purap.document.PurchaseOrderAmendmentDocument {
+	
+    protected String routeCode;
+    protected Building buildingObj;
+
+
+    /**
+     * Gets the routeCodeObj attribute. 
+     * @return Returns the routeCodeObj.
+     */
+    public Building getBuildingObj() {
+        return buildingObj;
+    }
+
+    /**
+     * Sets the routeCodeObj attribute value.
+     * @param routeCodeObj The routeCodeObj to set.
+     */
+    public void setBuildingObj(Building buildingObj) {
```

Comment:
Please add:
    if( ObjectUtils.isNull(buildingObj) ){
            this.refreshReferenceObject("buildingObj");
     }